### ISSUE TITLE
Ability to unlist Terminal buffer

### ISSUE THREAD

**Issue by dseeni** [coc-extensions/coc-powershell#97]
Is there anyway to close the powershell integrated console in neovim? It keeps on launching, even if I kill the buffer it automatically recreates itself.

**Comment by dseeni** [comment]
nevermind, you guys added coc hide terminal!!!!!!!!!!!!! 
hooray!!!!!!!!!!! coc hide terminal <3<3<3

**Comment by dseeni** [comment]
Reopening this issue
there is no way to close the integrated console buffer? it automatically recreates itself
![image](https://user-images.githubusercontent.com/6591429/79919788-e457de00-83e3-11ea-9c05-fa5710814087.png)

**Comment by TylerLeonhardt** [comment]
you can set this setting:

```
"powershell.integratedConsole.showOnStartup": true
```

or use the CocAction `powershell.toggleTerminal`

**Comment by dseeni** [comment]
Tyler, 
Neither CocAction OR showOnStartup solve this issue, if it were so I would be very pleased. 

Unfortunately that does NOT make the seperate powershell integrated console disappear. There is literally no way to close the buffer or quit. 

The only work around I have for this is setting up an autocmd to unlist the buffer, like so: 
![image](https://user-images.githubusercontent.com/6591429/80060295-10a65400-84e3-11ea-8a39-7ae0d2e5d710.png)

There is literally no way to get rid of the powershell buffer, the small window yes you can toggle hide it and set showOnStartup:False (Which I have done).. this works for the small window on startup, but not the seperate powershell buffer. 

Reading into it it seems this running live instance of the integrated console is necessary for Microsoft's LSP completion and documentation in Coc-Powershell. Which is fine, I like that stuff, so the unlisted buffer seems best, "out of sight out of mind" you know?

**Comment by TylerLeonhardt** [comment]
Like you said, all of the language features come from the Integrated Console if you get rid of that, then you don't have any of the goodness that coc-powershell provides.

The reason why the Integrated Console is displayed by default is because this extension is attempting to replicate the experience that a lot of PowerShell users expect:

* script on top
* REPL on bottom
* some ability to run parts of the script on top in the REPL below (via powershell.Execute CocAction)

This experience was made famous in the PowerShell ISE and the PowerShell extension for VS Code.

The CocAction to toggle and the settings were specifically added for the purpose of hiding it for those that don't want to see it but it's still something that needs to exist for language features.

Is there a reason this isn't good enough? I'm just trying to understand what your scenario is.

**Comment by dseeni** [comment]
I just like my buffer list to be buffers I want to be seeing. When you cycle through buffers, say I'm working with only 2 files, its nice to swap between them, rather than have a mandatory 3rd buffer (the powershell ISE) in between. Really just a preference thing. 

Of course, the CocAction to hide the split in the powershell file could also unlist the ISE buffer (basically my autocmd). And 'show' could list that separate buffer. Would basically be the same thing.  Anyway my workaround has me satisfied, I get all the benefits of Coc-Powershell without an unnecessary buffer in my bufferlist.

**Comment by yatli** [comment]
Yes I think this is a preference, and our current implementation complies to the nvim defaults (terms are listed). You can use something like [vmux|https://github.com/yatli/vmux.vim] to automatically unlist the terminal buffers.

**Comment by TylerLeonhardt** [comment]
Yeah we call to coc.nvim to create the terminal buffer and we call hide on it:

https://github.com/coc-extensions/coc-powershell/blob/ddbd0bd3c96a61131f08f9341d24f6daaec30765/src/client/process.ts#L144

I think that method could also unlist or maybe there can be an unlist API as well. @chemzqm do you have any thoughts here?

**Comment by chemzqm** [comment]
> could also unlist or maybe there can be an unlist API as well.

No, it just hide the buffer.  Use autocmd in your vimrc if you want to unlist terminal buffer.

**Comment by dseeni** [comment]
> No, it just hide the buffer. Use autocmd in your vimrc if you want to unlist terminal buffer, but looks like @dseeni want to destroy the buffer.

I ended up using an autocmd I posted above, unlist is fine. I don't mind that it's there, since it's required for powershell LSP to work. 

It's more of a subjective UX thing, I like to switch buffers without running into this integrated terminal buffer. 

I know there were several users who would like it to just be out of the way, commonly a user already has a terminal established. Unlist is fine. I appreciate all the responses!

**Comment by chemzqm** [comment]
My suggestion is use a subprocess to start language server, and use prompt buffer (which is quite new on neovim) for interactive.

**Comment by bgmonroe** [comment]
> Reopening this issue
> there is no way to close the integrated console buffer? it automatically recreates itself

You can close the REPL (only after closing the editor buffer unfortunately) by using the '**[Environment]::Exit(1)**' command.  But to be honest, it (the REPL buffer) should:

1. Be closable with 'exit', 'Exit', or ^d.  I think the current behavior is a side effect of how the pwsh process gets spun up by the extension. 
2. Close when the editor buffer closes. 
3. Be configurable to not open by default (I like that it opens but it should be user configurable with a simple bool var in the coc-settings.json).
